backend/api/projects.py

```diff
@@ -0,0 +1,187 @@
+                    json5.loads(request.data.decode('utf-8')))
+            else:
+                raise ApiException("Wrong content header" +
+                                   "and no files attached", 400)
+            return jsonify(return_ids)
+
+        except ApiException as e:
+            raise e
+
+        except UnicodeDecodeError as ue:
+            raise ApiException("Only utf-8 compatible charsets are " +
+                               "supported, the request body does not " +
+                               "appear to be utf-8.", 400)
+        except Exception as err:
+            raise ApiException(str(err), 400)
+
+
+@projects.route('/api/projects', methods=['GET'])
+def get_projects():
+    """Return list of projects, args->limit, skip
+
+    Returns:
+        res: A list of projects
+    """
+    limit = request.args.get('limit', type=int)
+    skip = request.args.get('skip', type=int)
+    argc = len(request.args)
+
+    if g.projects.count() == 0:
+        return make_response("There are no projects", 500)
+
+    if argc == 0:
+        res = g.projects.find({})
+    elif limit and skip and argc < 3:
+        res = g.projects.find({}, limit=limit, skip=skip)
+    elif limit and argc < 2:
+        res = g.projects.find({}, limit=limit)
+    elif skip and argc < 2:
+        res = g.projects.find({}, skip=skip)
+    else:
+        return make_response('Invalid parameters', 400)
+
+    res = make_response(jsonify([x for x in res[:]]))
+    res.headers['Content-Type'] = 'application/json'
+    return res
+
+
+@projects.route('/api/projects/<uuid:project_id>', methods=['GET'])
+def get_project_by_id(project_id):
+    """Returns project by ID number, 404 if it is not found.
+
+    Args:
+        project_id: The ID of the project which should get returned
+
+    Returns:
+        res (json): Project corresponding to the ID
+    """
+    res = g.projects.find_one({'_id': project_id})
+    if res is None:
+        return make_response("Project not found", 404)
+    return jsonify(res)
+
+
+@projects.route('/api/projects/<uuid:project_id>', methods=['DELETE'])
+# @roles_required('admin')
+def delete_project(project_id):
+    """Deletes a project by ID.
+
+    Args:
+        project_id: ID of a project
+
+    Returns:
+        response: Success response or 404 if project is not found
+    """
+    if g.projects.delete_one({'_id': project_id}).deleted_count == 0:
+        return make_response("Project could not be found", 404)
+    else:
+        return make_response("Success")
+
+
+@projects.route('/api/projects/<uuid:project_id>', methods=['PUT'])
+# @login_required
+def update_project(project_id):
+    """Updates Project by ID
+
+    Args:
+        project_id, updated manifest in json/json5 format
+
+    Returns:
+        response: Success response
+                  or 404 if project is not found
+                  or 409 if project_id differs from manifestID
+                  or 500 in case of validation error
```

Comment:
this should not return 500 (internal server error) in case of a validation error @Krisselchen can you fix this after this is in master?